### PR TITLE
Implemented Assign Tasks, and Added Task and TaskList class.

### PR DESCRIPTION
**Pull request by Harish-coding**
Implemented "assign" command. Closes issue #4.

Added AssignCommand, AssignCommandParser, Task, TaskList,
DuplicateTaskFoundException files to implement the command.

Additionally, included a private overloaded constructor for Person class
 for convenience.

Have to accordingly update implementation of "archive" command 
@Korochii.

Have to update test cases.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.exceptions.DuplicateTaskException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+import seedu.address.model.person.StudentId;
+import seedu.address.model.person.Task;
+
+
+/**
+ * Assigns a task to a person in address book.
```
Comment (Korochii) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:17]:
```suggestion
 * Assigns a task to a student in TAPA.
```

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Assigns a tasks to a student. "
+            + "Parameters: "
+            + PREFIX_ID + "STUDENT_ID "
+            + PREFIX_TASK_NAME + "TASK_NAME\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ID + "A0123456Z "
+            + PREFIX_TASK_NAME + "assignment 1";
+
+    public static final String MESSAGE_SUCCESS = "Task Assigned: %1$s";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task is already assigned to person.";
```
Comment (Korochii) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:32]:
```suggestion
    public static final String MESSAGE_DUPLICATE_TASK = "This task is already assigned to the specified student.";
```

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -58,6 +77,57 @@ public Email getEmail() {
         return email;
     }
 
+    public TaskList getTaskList() {
+        return taskList;
+    }
+
+
```
Comment (Korochii) [inline src/main/java/seedu/address/model/person/Person.java:84]:
```suggestion
```
Additional whitespace to be removed.

File: src/main/java/seedu/address/model/person/Person.java [hunk 4]
```diff
@@ -71,6 +141,11 @@ public boolean isSamePerson(Person otherPerson) {
                 && otherPerson.getStudentId().equals(getStudentId());
     }
 
+    public Person getCopy() {
```
Comment (Korochii) [inline src/main/java/seedu/address/model/person/Person.java:144]: Might be good if we can include a JavaDoc header here

File: src/main/java/seedu/address/model/person/Task.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return taskName;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+
```
Comment (Korochii) [inline src/main/java/seedu/address/model/person/Task.java:133]:
```suggestion
```

File: src/main/java/seedu/address/model/person/TaskList.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+     * @param task the task to be added.
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+    }
+
+    /**
+     * Adds all tasks from an existing taskList to the list of tasks.
+     *
+     * @param taskList the target taskList to be copied from.
```
Comment (Korochii) [inline src/main/java/seedu/address/model/person/TaskList.java:42]:
```suggestion
     * @param taskList the target TaskList to be copied from.
```

**Review by Korochii — COMMENTED** [review]
Generally looks good to me! (Other than a few minor formatting issues which should be easy to fix...)

In addition, the errors that are highlighted by the failing CI should be fixed before we merge too. [[Relevant link for trailing whitespaces](https://github.com/nus-cs2103-AY2122S2/forum/issues/197)]

@AY2122S2-CS2103T-W09-4/developers Would be great if someone else review this PR too (since I kinda wrote some of the code here...)

File: src/main/java/seedu/address/model/person/TaskList.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+/**
+ * Maintains the list of tasks assigned to a student.
+ */
+public class TaskList {
+
+    private final ArrayList<Task> taskList;
+
+    /**
+     * Constructs a {@code Task}.
```
Comment (Korochii) [inline src/main/java/seedu/address/model/person/TaskList.java:15]:
```suggestion
     * Constructs a {@code TaskList}.
```

File: src/main/java/seedu/address/logic/commands/AssignCommand.java (hunk 1 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java]: Added suggestion! 👍

File: src/main/java/seedu/address/logic/commands/AssignCommand.java (hunk 2 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java]: added suggestion! 👍

File: src/main/java/seedu/address/model/person/Person.java (hunk 3 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/model/person/Person.java]: Added suggestion! 👍

File: src/main/java/seedu/address/model/person/Person.java (hunk 4 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/model/person/Person.java:148]: added JavaDocs comment! 👍

File: src/main/java/seedu/address/model/person/Task.java (hunk 5 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/model/person/Task.java]: Added suggestion! 👍

File: src/main/java/seedu/address/model/person/TaskList.java (hunk 6 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/model/person/TaskList.java]: Added suggestion! 👍

File: src/main/java/seedu/address/model/person/TaskList.java (hunk 7 above)
Comment (Harish-coding) [reply src/main/java/seedu/address/model/person/TaskList.java]: Added suggestion! 👍

**Review by medajet — COMMENTED** [review]
LGTM, other than a minor nitpick in the Task class.

File: src/main/java/seedu/address/model/person/Task.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    private final String taskName;
+    private boolean isComplete;
+
+    /**
+     * Constructs a {@code Task}.
+     * Default constructor, to be used only by {@code JsonAdaptedPerson}.
+     */
+    public Task() {
+        super();
```
[the remark this answers is not in the document]
Comment (medajet) [reply src/main/java/seedu/address/model/person/Task.java:32]:
```suggestion

```

[the remark this answers is not in the document]
Comment (Korochii) [reply src/main/java/seedu/address/model/person/Task.java:32]: Hi Jet, thanks for pointing that out! 
Initially, the super() line was included as I was referring to this article. ([3. JsonMappingException: No Suitable Constructor](https://www.baeldung.com/jackson-exception))
However, after further testing, I realised that TAPA will still work fine without this line here.

Will be committing this suggestion 😃

**Review by medajet — APPROVED** [review]
LGTM! 🙂

**Pull request merged**